File: Desktop/sanofood/src/Entity/Conseil.php
<?php
// src/Entity/Conseil.php
namespace App\Entity;

use Symfony\Component\Validator\Constraints as Assert;
use Doctrine\ORM\Mapping as ORM;
use App\Repository\ConseilRepository;
#[ORM\Entity(repositoryClass: ConseilRepository::class)]
class Conseil
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(type: 'integer')]
    private ?int $idC = null;

    #[ORM\Column(type: 'string', length: 255)]
    #[Assert\NotBlank(message: 'Le contenu ne doit pas être vide.')]
    #[Assert\Length(
        max: 255,
        maxMessage: 'Le contenu ne doit pas dépasser {{ limit }} caractères.'
    )]
    private ?string $contenu = null;

    #[ORM\Column(type: 'datetime')]
    #[Assert\NotBlank(message: 'La date du conseil est obligatoire.')]
    #[Assert\Type(
        type: \DateTimeInterface::class,
        message: 'La date doit être au format valide (YYYY-MM-DD HH:MM:SS).'
    )]
    #[Assert\GreaterThanOrEqual(
        value: 'today',
        message: 'La date du conseil ne peut pas être antérieure à aujourd\'hui.'
    )]
    private ?\DateTimeInterface $dateConseil = null;

    #[ORM\ManyToOne(targetEntity: Demande::class, inversedBy: 'conseils')]
    #[ORM\JoinColumn(name: 'id_demande', referencedColumnName: 'id')]
    private ?Demande $demande = null;

    // Getters et Setters
    public function getIdC(): ?int
    {
        return $this->idC;
    }

    public function getContenu(): ?string
    {
        return $this->contenu;
    }

    public function setContenu(string $contenu): self
    {
        $this->contenu = $contenu;
        return $this;
    }

    public function getDateConseil(): ?\DateTimeInterface
    {
        return $this->dateConseil;
    }

    public function setDateConseil(\DateTimeInterface $dateConseil): self
    {
        $this->dateConseil = $dateConseil;
        return $this;
    }

    public function getDemande(): ?Demande
    {
        return $this->demande;
    }

    public function setDemande(?Demande $demande): self
    {
        $this->demande = $demande;
        return $this;
    }
}

File: Desktop/yahya123/src/main/java/entities/Conseil.java
package entities;

import java.time.LocalDateTime;

public class Conseil {
    private int id;
    private String contenu;
    private LocalDateTime dateConseil;
    private int idDemande;

    public Conseil() {}

    public Conseil(int id, String contenu, LocalDateTime dateConseil, int idDemande) {
        this.id = id;
        this.contenu = contenu;
        this.dateConseil = dateConseil;
        this.idDemande = idDemande;
    }

    // Fixed constructor
    public Conseil(int conseilId, int idDemande, String content, LocalDateTime dateTime) {
        this.id = conseilId;
        this.idDemande = idDemande;
        this.contenu = content;
        this.dateConseil = dateTime;
    }

    public int getId() { return id; }
    public void setId(int id) { this.id = id; }

    public String getContenu() { return contenu; }
    public void setContenu(String contenu) { this.contenu = contenu; }

    public LocalDateTime getDateConseil() { return dateConseil; }
    public void setDateConseil(LocalDateTime dateConseil) { this.dateConseil = dateConseil; }

    public int getIdDemande() { return idDemande; }
    public void setIdDemande(int idDemande) { this.idDemande = idDemande; }
}